### DISCUSSION TITLE
Teleport with Openssh nodes, multiple CAs

### DISCUSSION THREAD

**Discussion by shark-low** [gravitational/teleport#8040] [Q&A] [answered]
Hi there,

I'd like to have two instances of Teleport proxies, with two different configurations let's call them A & B. Nodes are Openssh's sshd, not Teleport.
Doing all the procedure indicated in the docs is working all right (using tsh or native ssh client). 
Adding both the A & B user CAs into the same file is working perfectly fine, as stated in the manpage.

But, when I'm adding the two HostCertificate and the two Hostkey generated on A & B into sshd_config, only the first is effective.
If I'm logged in proxy A and the HostKey from B is the first in the config file, I can't login and "vice et versa". But in the manpage it's clearly indicated I can load multiple HostKey . And in sshd's logs I can see they're loaded (as the certs too).

Message is (Teleport side): 
`ERROR: ssh: rejected: connect failed (ssh: handshake failed: ssh: no authorities for hostname: **redacted**:0)`

Except a disconnected session there's nothing in sshd's logs.

Is there some way to run Teleport with Openssh nodes configured with two different Host CAs in the same config ??? I'd really like to avoid sharing the same config running etcd.

Thanks for any help

**Comment by webvictim** [comment, marked as answer]
This may be a bug where Teleport is only checking the first host certificate presented for a given hostname. With that said, your use case isn't really supported - is there some way you could leverage [trusted clusters](https://goteleport.com/docs/trustedclusters/) instead? Can you explain what the logic is behind having the same node available from two different Teleport clusters with the same hostname?

**Reply by shark-low** [reply]
Hey thanks !

Yes indeed, trusted clusters was the answer, now I'm good (at least looks like it atm). I can login with both A & B on my openssh servers. I don't think it's a teleport bug but more something to do with the Openssh's sshd. The only workaround I could find was to generate hostkeys with different crypto and specify this on the cmdline when using openssh client to select which HostKey will be used.

About the use-case, I need some sort of backup server in case of failure and I have management who doesn't want to run a shared configuration with etcd (like explained in HA chapter) and want to use HSMs to protect the private CA keys. I'm not well versed into HSM yet but it looks like a private CA is generated and then "sealed" into the HSM without exporting capability, how does one can share a configuration using HSMs (the yubi one) ? As you can see there's no logic, only ignorance...

If I may, documentation is very weird, I mean there are lots of stuff in there, but it's not always very clear to the reader. How to do things is relatively well explained, different function of each component and why to use it is much more complicated to grasp (except for cloud users). Add a pinch of reverse proxy and docker stuff and you're almost lost...
 I thought I'll find more information into the videos (and I hate video doc for computer related stuff) just to find that it was only read out loud documentation for millenials refusing to read text (sorry... salty :)). I mean you can actually see the dude searching for the next line in the written documentation to continue his clip; what's the point of that (blind people accessibility ?). 

I don't want to finish on a bad remark, putting this doc issue aside, this Teleport product is really something interesting and thanks for all the hard work people are putting in it.

**Reply by webvictim** [reply]
Glad you got things working. Thank you for the comments - we appreciate all candid feedback :)

If you do have any ideas regarding what would you’d find more useful in the documentation, we’d be happy to hear them and consider improvements.
